File: notes_api_service/src/constants/domains/user_constants.nim
import json
from ../../domains/user/infrastructure/entities/user_entity import UserEntity

const USER_ID_KEY* = "id"

const USER_USERNAME_KEY* = "username"

const NOT_EXISTING_USER_EXCEPTION_MESSAGE*: string = "User Does Not Exist!"

const ALREADY_EXISTING_USER_EXCEPTION_MESSAGE*: string = "User Already Exists!"

const USER_ID* = 0

const USER_USERNAME* = "Samuel de Carvalho"

let USER_JSON* = %*{ USER_USERNAME_KEY: USER_USERNAME }

let USER_DATA_TRANSFER_OBJECT_JSON* = %*{ USER_ID_KEY: USER_ID, USER_USERNAME_KEY: USER_USERNAME }

let LIST_USER_DATA_TRANSFER_OBJECT_JSON* = %*[{ USER_ID_KEY: USER_ID, USER_USERNAME_KEY: USER_USERNAME }]

let LIST_OF_USERS_JSON* = %*[{ USER_USERNAME_KEY: USER_USERNAME }]

let USER_JSON_WITH_WRONG_DATA* = %*{ USER_USERNAME_KEY: "Other Username" }

let USER_OBJECT* = UserEntity(id: USER_ID, username: USER_USERNAME)

let USER_OBJECT_WITH_WRONG_DATA* = UserEntity(id: USER_ID, username: "Other Username")

let MOST_USER_ID* = 999999999

File: in_memory_cache_service/src/constants/domains/user_constants.ts
export const USER_IN_MEMORY_DATABASE_KEY = "users";

export function USER_HASH_IN_MEMORY_DATABASE_KEY(id: number): string {
  return `user:${id}`;
}

export const NOT_EXISTING_USER_ERROR_NAME = "User Does Not Exist Error";

export const NOT_EXISTING_USER_ERROR_MESSAGE = "User Does Not Exist";

export const ALREADY_EXISTING_USER_ERROR_NAME = "User Already Exists Error";

export const ALREADY_EXISTING_USER_ERROR_MESSAGE = "User Already Exists";

export const USER_ID_IN_MEMORY_DATABASE_KEY = "id";

export const USER_USERNAME_IN_MEMORY_DATABASE_KEY = "username";

export const USER_ID = 0;

export const USER_USERNAME = "Samuel de Carvalho";
